core/src/main/scala/org/apache/spark/network/netty/SparkTransportConf.scala

```diff
@@ -44,8 +44,11 @@ object SparkTransportConf {
     // assuming we have all the machine's cores).
     // NB: Only set if serverThreads/clientThreads not already set.
     val numThreads = NettyUtils.defaultNumThreads(numUsableCores)
-    conf.setIfMissing(s"spark.$module.io.serverThreads", numThreads.toString)
-    conf.setIfMissing(s"spark.$module.io.clientThreads", numThreads.toString)
+    val finalServerNumThreads = getNumOfThreads(conf, module, true, numThreads)
+    val finalClientNumThreads = getNumOfThreads(conf, module, false, numThreads)
+
+    conf.setIfMissing(s"spark.$module.io.serverThreads", finalServerNumThreads)
```

Comment:
no, it's opposite. spark.driver.$module.io.serverThreads takes precedence over spark.$module.io.serverThreads. So does spark.executor.* . 

The below code is to set spark.$module.io.serverThreads to driver's value or executor's value, if specified, to avoid downstream changes of retrieving serverThreads from different configuration items.
` conf.setIfMissing(s"spark.$module.io.serverThreads", finalServerNumThreads)`

Reply:
From my understanding, if `spark.$module.io.serverThreads` is configured, `finalServerNumThreads` will not take effect, since the config is only set if missing `setIfMissing`. Please correct me if I'm wrong.

Reply:
You are correct. sorry for the confusion. The intention is to give user a chance to use driver's specific value and executor's specific value. If the specific values are given, spark.$module.io.serverThreads should not be provided.  spark.$module.io.serverThreads should be missing.

Reply:
@jerryshao , After some discussion within team, we think io threads should be override by the specific values if both values are provided. I just made one more commit. please help review.

thanks.